File: mahaam-api-py/infra/monitor/monitor_models.py
from uuid import UUID
from datetime import datetime
from pydantic import BaseModel, Field


class Traffic(BaseModel):
    id: UUID
    health_id: UUID = Field(serialization_alias="healthId")
    method: str
    path: str
    code: int | None = None
    elapsed: int | None = None
    headers: str | None = None
    request: str | None = None
    response: str | None = None


class TrafficHeaders(BaseModel):
    user_id: UUID | None = Field(default=None, serialization_alias="userId")
    device_id: UUID | None = Field(default=None, serialization_alias="deviceId")
    app_version: str | None = Field(default=None, serialization_alias="appVersion")
    app_store: str | None = Field(default=None, serialization_alias="appStore")


class Health(BaseModel):
    id: UUID
    api_name: str = Field(serialization_alias="apiName")
    api_version: str = Field(serialization_alias="apiVersion") 
    node_ip: str = Field(serialization_alias="nodeIp")
    node_name: str = Field(serialization_alias="nodeName")
    env_name: str = Field(serialization_alias="envName")
    started: datetime | None = None
    pulse: datetime | None = None
    stopped: datetime | None = None
    
    model_config = {"populate_by_name": True, "use_enum_values": True}

File: mahaam-api-cs/Src/Infra/Monitor/MonitorModels.cs
namespace Mahaam.Infra.Monitoring;

public class Traffic
{
	public Guid Id { set; get; }
	public Guid HealthId { set; get; }
	public required string Method { set; get; }
	public required string Path { set; get; }
	public int? Code { set; get; }
	public long? Elapsed { set; get; }
	public string? Headers { set; get; }
	public string? Request { set; get; }
	public string? Response { set; get; }
}

public class TrafficHeaders
{
	public Guid? UserId { set; get; }
	public Guid? DeviceId { set; get; }
	public string? AppVersion { set; get; }
	public string? AppStore { set; get; }
}

public class Health
{
	public Guid Id { get; set; }
	public string? ApiName { get; set; }
	public string? ApiVersion { get; set; }
	public string? NodeIP { get; set; }
	public string? NodeName { get; set; }
	public string? EnvName { get; set; }
}
